packages/perseus/src/util/snowman-utils.test.ts

```diff
@@ -0,0 +1,120 @@
+import {
+    addWidget,
+    getAllWidgetIds,
+    WidgetType,
+    getQuestionWidgetIds,
+    getAllWidgetTypes,
+} from "./snowman-utils";
+
+describe("widgetRegex", () => {
+    it("locates match and capture groups as expected for non-hyphenated widget types", () => {
+        // Arrange
+        const exampleContent = "Here is some content [[☃ group 1]]";
+        const widgetRegex = /\[\[☃ (([a-z-]+) \d+)\]\]/g;
```

Comment:
For sure. I initially did it this way because the regex is stateful, and that persists. I've written out two potential ways to fix this though. It looks like most instances in the code of using this regex use it in a while loop, which resets the lastIndex property of the regex to 0 once the search finishes. It can also be reset manually by setting .lastIndex to 0. I'll go over these during our group pair time today.